firebase-firestore/src/main/java/com/google/firebase/firestore/index/FirestoreIndexValueWriter.java

```diff
@@ -138,6 +142,22 @@ private void writeUnlabeledIndexString(
     encoder.writeString(stringIndexValue);
   }
 
+  private void writeIndexVector(MapValue mapIndexValue, DirectionalIndexByteEncoder encoder) {
+    writeValueTypeLabel(encoder, INDEX_TYPE_VECTOR);
+
+    Map<String, Value> map = mapIndexValue.getFieldsMap();
+
+    // Vectors sort first by length
+    String key = Values.VECTOR_MAP_VECTORS_KEY;
+    int length = map.get(key).getArrayValue().getValuesCount();
+    this.writeValueTypeLabel(encoder, INDEX_TYPE_VECTOR);
```

Comment:
This was a typo. The value type label was updated to number, which is the type of the data being written.

Reply:
This could not be evaluated in the pre-existing tests, because there are no current vector implementations where the vector size would be represented by any other data type. This was likely to never cause an issue, but it is corrected now. And I added a new style of test to assert that `FirestoreIndexValueWriter.writeIndexValue` generates indexes as expected.

Reply:
cool, Thanks!